### DISCUSSION TITLE
Label Generator Feedback

### DISCUSSION THREAD

**Discussion by hay-kot** [hay-kot/homebox#273] [General]
Future home of label generator feedback

**Comment by calrock27** [comment]
Hi there - Just wanted to say that I bookmarked this project awhile back and finally got the bandwidth to begin implementation. It's awesome that you've just rolled out the QR/Label functionality. The overall quality and functionality of HomeBox is incredible for my home use case.

I know that you've built the Label Generator feature as "intended to be print-ahead labels" and I was wondering what your thoughts were on having it display/print existing asset data for the asset IDs that have already been populated? This would allow me/users to print/reprint fully prepared asset labels rather than hand write the labels.

Alternatively, where you're working on #16  I think I may be able to accomplish this myself in label making software if the export would include the permanent HomeBox link. I'm not sure if the database export would include the permanent asset links though.

Thank you for all of your work on this, it's truly an incredible project.

**Reply by hay-kot** [reply]
I'm almost done with full export support to CSV. Once that's done it should make it super simple to build your own tool to generate labels into a specific format. The as long as the asset ID is known you can calculate the label path `https://domain.example.com/a/{asset-id}` will get you there.

**Reply by calrock27** [reply]
Excellent! Thanks for the response. Very excited to begin importing my assets. Currently in the process of inventorying all of my belongings.. Slowly but surely.

**Comment by stegl** [comment]
Hello everyone
it would be nice to have option to set how many labels is in row (with this you can print it on labels like this
![p_1200x1200](https://user-images.githubusercontent.com/65857447/219701901-3e8d6861-ca03-450c-9bb7-9ad5876ca7a3.jpg) )

also if you scan qr code it automatically opens up item in homebox. I dont see a need to write anything to label... option to only have qr code maybe? 

//offtopic -> it would be nice to have option to print out locations (bulk)... it could use same system as here (i have about 300 of locations - im migrating from partkeepr to homebox)


other than that it is fine and it all works as it should. Did a test print today and it looks good but i have to cut it ( A4 with 4 labels)

Thank you very much for this amazing app. Great work.

Have a nice day.

**Reply by djmixman** [reply]
I use both a Brother and Dymo label printer. It would be awesome to support those.

**Reply by stegl** [reply]
I found out my label printer software supports function to +1 on every label.. So i set how number looks and then it automatically every time adds +1 (000,001,002,003...) take a look in software maybe you have this function too.

**Reply by estranged-networks** [reply]
hi, is it possible to share the settings for printing to a roll of labels like this picture above , rather than an A4 sheet of paper. ? thanks in advance - 

**Edit: update**
Hi again, I have managed to get the right settings for printing to smaller labels - you are right it is trial and error but it doesnt have to be so, May I suggest ...

1. Add a unit of messurement so that its not guess work ( inches, cm, mm anything would be great ) 
2. Allow to save the template page because everytime you go to print it defaults back to A4 settings
3. Include the last label in the print - currently if I select labels 1-10 it will display & print 1-9 
4. Allow customisation of the label - Homebox & the two lines, even the preceeding 000- ,  its just wasted ink in my opinion. 
5. Be able to print/re-print a single lable from its' assest page, be it the assest or the location label ( I think others have also shown an interest for this as a feature )

Thank you thus far everything else is amazing , just printing needs some ❤️

**Reply by danielrosehill** [reply]
I'm using a Brother too (specifically the QL 700 -  amazing machine). I've been generating DataMatrix barcodes from the item URLs instead of using the default QR codes because I find the read performance a bit better. Not too much work to do it all manually but yes .. any templates that support the various standard continuous roll lengths would be a nice addition!

**Comment by internallabworks** [comment]
Hi!

This project is exactly what I've been looking for.
Only thing is, the Avery 5260 labels are not readily available in Europe it seems.
Would it be possible to add support for Avery 3475 which is readily available for cheap here in Europe?

**Reply by hay-kot** [reply]
Is there any reason you can use the current configuration to adjust the layout to support those labels?

The main issue with supporting other formats is it's mostly trial and error, so I can't add more unless I have the labels. If you need more knobs to adjust in the UI to get the 3475 labels to work then we could add some more configuration options. 

![CleanShot 2023-02-25 at 11 29 09](https://user-images.githubusercontent.com/64056131/221378290-8628a024-e4b6-4523-9d31-170350973721.png)

**Reply by internallabworks** [reply]
I tried adjusting page size to A4 size, and added the size of the labels.
For some reason it would not put 3 per line only 2 even though the label sheet has 24 labels on one page.
labels are slightly bigger than the 5260 labels but fit on a standard A4 sheet:
![3475_5014702815195_line](https://user-images.githubusercontent.com/64136358/221378625-a2b00238-8537-4a7a-866b-b7ca393f66fe.jpg)

**Reply by internallabworks** [reply]
I think I managed to "fix" it by using excel and the avery online app.
Will need to buy the labels and testprint, but it looks promising so far

**Reply by internallabworks** [reply]
Hmm, I think this project is not working as I thought.
I had plans to put a label on a storage box, then have the box registered in Homebox and then register the contents of the box.
Putting a label on each and every thing is not the way to find things imho.
Would it be possible to add some kind of box-id, so we could create labels for boxes too?

**Reply by hay-kot** [reply]
I'm not opposed to that, but It's not currently supported. You can use the QR code from a specific location to manually create labels for those areas in the mean time.

![image](https://user-images.githubusercontent.com/64056131/221430989-ec032d91-7a41-4dd9-ab1a-2aaa015e49fd.png)

**Reply by lennon101** [reply]
+1 for this feature. I would like the option to print only the QR codes for areas. This is how I would find items: scanning a qr code on a box in the garage filled with items. I'm not really sure what the point or use-case would be having a QR code for every single item and why you would be scanning it?

**Reply by Bumpalert** [reply]
I found that it is an inconvenience to restore all %3A%2F%2F by ://  and click the link after generating qr code. I generates the qr code outside homebox due to this issue.

**Reply by psalkowski** [reply]
I'm looking into inventory system for home usage and also would love to have labels on "locations" generated in similar way to items. Maybe we could add Asset ID to locations as well? For example, in different format, like `A-000-001` (and legacy `000-000`) is for assets, but for `L-000-000` is for locations

**Reply by JonathanGawrych** [reply]
I'm about to move, so I'm also scratching my head at this situation. I thought of a convoluted way to associate asset ids with locations (creating items with asset ids representing the location, inside the location. i.e. "box of cables" 000-001 item inside a "box of cable" location), but in the end I don't think it is worth it. I want bigger QR codes (4" rather than 1", with lines at the bottom rather than side), so I think I will have to do it manually anyways.

**Comment by LinusZhang5776** [comment]
Can I manually change the text 'homebox' in the label to other text? I want to create labels for several locations, but I don't want to handwrite this label.

**Reply by GeorgeHahn** [reply]
It would be awesome if there were a way to generate labels for assets. Rather than printing a bunch of blank labels, matching up the numbers to assets, and then writing in a description, you could create assets, print their labels, and place them accordingly.

**Reply by cRaZy92** [reply]
+1, I would love to see an option to include the item name in the labels.

**Comment by bernouilli90** [comment]
Hi! Could you add the option to adjust the padding between the labels (vertically and horizontally)? Could you add the option to change the measure unit to milimeters?

**Reply by pete1450** [reply]
To reference my current problem; I'd like to have really small labels and include the text/lines as well but they seem to be static. Once I shrink the QR code enough the text starts intersecting with the lines above. Having them be dynamic/adjustable would be great and padding would take care of the intersections.

**Comment by timdonovanuk** [comment]
Just seen this discussion - currently the label-generator doesn't seem to generate URLs properly. https://github.com/hay-kot/homebox/issues/564

**Comment by davegershgorn** [comment]
Hi! It would be great if scanning a QR code that isn't associated with an existing ID creates that asset, rather than returning an error that the asset isn't found. Thanks!

**Reply by ranasats** [reply]
Yes, this seems like how this feature should work, but it doesn't. Would be the best way to implement this! Kind of counter-productive to go to an item and change it's asset ID to match the QR code. Instead scanning the code should create a new asset you can populate.

**Comment by TheGreyDiamond** [comment]
I know this has been discussed in comments before, however, I would want to make this a more "top-level" entry. 
It would be great if you could generate printable QR code labels for locations.

**Comment by Jkweks** [comment]
So far love the home box system for my use. If it would be possible to add an option to the label generator to only show a QR code that would be great. For home use a big label isn't always practical and even a 1x1 label with the QR code would be awesome. Currently if you set the size to 1x1 all you get is homebox and then lines to fill out manually which is kind of counter intuitive to my use. 

If I get time I may tinker with that code to see what I can do with my albeit limited
Coding knowledge.

**Comment by victorhooi** [comment]
I know the current Homebox label implementation is around "print-ahead" labels - where you print out say a sheet of 20 labels, and then back-fill in things afterwards.

However, it would be cool if Homebox could also support those of us with dedicated label printers (e.g. Brady, Brother P-Touch line etc.) where you can just print labels on demand, and then slap those on an asset.

For example, I have the [Brother PT-P750W](https://www.brother.com.au/en/labellers/all-labellers/pt-p750w) desktop label printer:

![image](https://github.com/hay-kot/homebox/assets/243499/184cc450-11a6-4eaf-864d-da8e3a1746bf)

It prints (and automatically cuts) nice little sticky labels you can apply directly to items - super convenient for labelling things.

Most other brands (Dymo, Rhino, Brady) have similar office/industrial label printers.

Would it be feasible to add an option in Homebox to either print a single label on demand, or even just bulk-select (e.g. checkbox/filter) some objects, and spool print labels for all of them to an installed label printer?

For example, the [Inventree](https://inventree.org) project has support for printing direct to label printers:

https://docs.inventree.org/en/latest/extend/machines/label_printer/

Here's their plugin for the Brother label printers:

https://github.com/inventree/inventree-brother-plugin

What do you think?
